src/components/FlowRunTimeline.vue

```diff
@@ -0,0 +1,77 @@
+<template>
+  <div class="flow-run-timeline-container">
+    <span class="flow-run-timeline-container__beta-badge">
+      Beta
+    </span>
+    <FlowRunTimeline
+      v-if="graphData.length > 0"
+      :graph-data="graphData"
+      :is-running="isRunning"
+    />
+  </div>
+</template>
+
+<script lang="ts" setup>
+  import { FlowRunTimeline, TimelineNodeData } from '@prefecthq/graphs'
+  import { useSubscription } from '@prefecthq/vue-compositions'
+  import { computed } from 'vue'
+  import { useWorkspaceApi } from '@/compositions'
+  import { FlowRun, TimelineNode } from '@/models'
+
+  const props = defineProps<{
+    flowRun: FlowRun,
+  }>()
+
+  const isRunning = computed(() => {
+    return props.flowRun.state?.name.toLowerCase() === 'running'
+  })
+
+  const api = useWorkspaceApi()
+
+  const graphSubscription = useSubscription(
+    api.flowRuns.getFlowRunsTimeline,
+    [props.flowRun.id],
+    { interval: isRunning.value ? 5000 : undefined },
+  )
+
+  function isValidTimelineNode(value: TimelineNode): value is TimelineNodeData {
+    return typeof value === 'object'
+      && 'startTime' in value
+      && value.startTime instanceof Date
+  }
```

Comment:
Should have mentioned this earlier. Generally type guards live next to their types. So sometime you might want to move this to wherever TimelineNode is defined.